cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/DataMigrator.java

```diff
@@ -67,38 +102,44 @@ public boolean isMigrationComplete() {
     while (resolutions.hasNext()) {
       int resolution = resolutions.next();
       MetricsTableMigration metricsTableMigration = migrationOrder.get(resolution);
-      if (metricsTableMigration.isOldMetricsDataAvailable()) {
+      if (metricsTableMigration.v2MetricsTableExists(hBaseTableUtil, resolution) &&
+        metricsTableMigration.isOldMetricsDataAvailable()) {
```

Comment:
if the table is not available, it would be better to skip doing scan to check if metrics data is available as that will throw an exception, so decided to do that only if table exists and remove the entry if the table no longer exists.
i also noticed that i wasn't handling possible exception say the table gets deleted while doing this check or any other exception during data transfer, i have added catch block to handle that currently.